pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -289,6 +291,9 @@ public void runJob() {
 
               int numRows = dataTable.getNumberOfRows();
               for (int rowId = 0; rowId < numRows; rowId++) {
+                if (rowId % MAX_ROWS_UPSERT_PER_INTERRUPTION_CHECK == 0 && Thread.interrupted()) {
+                  throw new EarlyTerminationException();
```

Comment:
nit: perhaps add an error message here?

Reply:
Yeah as discussed below, IIUC currently no one can really catch that error. But in the future we need to add handling code here if this thread is to be canceled explicitly.